src/runtime/admin/component-manager.ts

```diff
@@ -16,13 +18,48 @@ interface AddToStackWindowEvent {
 interface AddToStackEvent extends _ResourceStackItem, AddToStackWindowEvent {}
 
 export default class ComponentManager {
-  private lastClickTarget: HTMLElement|undefined
+  private lastClickTarget: HTMLElement | null = null
   private currentResourceStack: ResourceStackItem[] = []
+  private isEditing = false
 
-  public resetStack () {}
+  public setEditMode (newEditModeStatus: boolean) {
+    this.isEditing = newEditModeStatus
```

Comment:
As discussed in the PR, having a local variable for `isEditing` does not sit well with me when there is a global variable as we will need to make sure this stays in sync. Better to use the variable available in Pinia store.

Reply:
for me it looks like all editing-related stuff can be moved from `Admin` class to `ComponentManager` class, since it is responsible for stack operations. I don't know further plans, but currently it looks like it will be a good move

Reply:
Sorry but I disagree. And the Admin class is not holding the isEditing state. This is read and managed from a Pinia store.

Reply:
You have 2 options to get the isEditing state passed through

Passing the `adminStroeDefinition` and having another private get function for `adminStore` like the Admin class.

Or `storeToRefs` function from pinia to immediately use the store and pass the `isEditing` variable as a ref.

Reply:
Reasoning is, `isEditing` is a state of administration, and is relevant for many other services and functions. The Admin class is the service which exposes functions and values for our global state. It include navigation guards, UI changes so far an more in future.

`ComponentManager` is specific to the component managing functionality which is a result of being in an editable state. Other things will also be happening as a result of being in an editable state that are not related to the functionality of the component manager directly.

Reply:
Done in 572068ed3774ba788216ca1dab07f64ce7e32620
What I was trying to say is that since we refer to admin store within Component Manager class, in order to use store and isEditing status we define also the same getters for isEditing and store itself as we do in Admin class